</br>
<div class="row clearfix">
            <div></div>
                <div class="col-md-12 column">
                        <ul class="breadcrumb">
                                <li>
                                        <a href="index.php">Home</a> <span class="divider">/</span>
                                </li>
                                <li>
                                        <a href="#">Training</a> <span class="divider">/</span>
                                </li>
                                <li class="active">
                                        Workshops
                                </li>
                </ul>
            </div>
</div>

<div id="middle100"><!--middle start -->
<div id="middle"><!--middle -->
</br>
<h3>Six weeks sequence analysis bootcamp</h3>
</br>

<h2>
    Week 1: Unix environment
</h2>
<p>
    Goals:
</p>
<ol start="1">
    <li>
        Basic command line familiarity
    </li>
    <li>
        Changing standard input and output. Unix pipes.
    </li>
    <li>
        File compression tools: tar and gzip
    </li>
    <li>
        Introduction to some well-known unix tools: more, less, grep etc.
    </li>
    <li>
        Case studies on fastq files and sam files. For example, how do we get only the nucleotide sequences in a fastq file. How do we get entries having a particular adapter and etc.
    </li>
    <li>
        Understand how to transfer data across servers
    </li>
    <li>
        Network protocols: SSH, FTP, SFTP, SCP, CP etc
    </li>
    <li>
        How to customize bash, define paths and etc.
    </li>
    <li>
        Unix processes. How to kill a process, send it to background / foreground.
    </li>
</ol>
<h2>
    Week 2: Databases, annotations, genomic resources
</h2>
<p>
    Goals:
</p>
<ol start="1">
    <li>
        Understand basic file formats (FASTA, FASTQ , BED, GTF, SAM, BAM, WIG, BIGWIG, BEDGRAPH)
    </li>
    <li>
        Genomic resources and databases: UCSC, ENSEMBLE, NCBI
    </li>
    <li>
        Basic manipulation and visualization of these files:
    </li>
</ol>
<ol start="1">
    <li>
        Overlaps, intersections, etc
    </li>
    <li>
        Enrichment computations?
    </li>
</ol>
<ol start="4">
    <li>
        File conversions: bedgraph to wig, Wig to bigWig etc, sam to bam etc.
    </li>
</ol>
<h2>
    Week 3: The Galaxy environment
</h2>
<p>
    Goals:
</p>
<ol start="1">
    <li>
        Getting data in/out (with ftp and uploading from UI)
    </li>
    <li>
        Doing analysis in Galaxy
    </li>
</ol>
<h2>
    Week 4: Putting all together step 1: RNA-Seq data processing
</h2>
<p>
    Goals:
</p>
<ol start="1">
    <li>
        Understanding sequence data
    </li>
    <li>
        Transferring and manipulating RNA-Seq data
    </li>
    <li>
        Full data processing
    </li>
    <li>
        Visualization of processed RNA-Seq data
    </li>
</ol>
<p>
    Dataset:
</p>
<h2>
    Week 5: Introduction to R
</h2>
<p>
    Goals:
</p>
<ol start="1">
    <li>
        Understanding R-Studio online
    </li>
    <li>
        Basic data analysis
    </li>
    <li>
        How to install and use packages locally
    </li>
    <li>
        A glimpse of bioconductor?
    </li>
</ol>
<p>
    Datasets:
</p>
<h2>
    Week 6: Putting all together step 2: RNA-Seq data analysis
</h2>
<p>
    Goals: Using R to analyze RNA-Seq data
</p>
<ol start="1">
    <li>
        Normalizing data
    </li>
    <li>
        Clustering expression data
    </li>
    <li>
        Gene ontology analysis
    </li>
    <li>
        Differential expression
    </li>
    <li>
        principal component analysis
    </li>
</ol>

</div>
</div>
